jopa-impl/src/main/java/cz/cvut/kbss/jopa/proxy/lazy/LazyLoadingProxyFactory.java

```diff
@@ -59,8 +61,18 @@ public <T> Object createProxy(T entity, FieldSpecification<? super T, ?> fieldSp
         } else if (Map.class.isAssignableFrom(type)) {
             return new LazyLoadingMapProxy<>(entity, (FieldSpecification) fieldSpec, uow);
         } else if (Collection.class.isAssignableFrom(type)) {
-            final PluralAttribute<? super T, ?, ?> pa = (PluralAttribute<? super T, ?, ?>) fieldSpec;
-            return switch (pa.getCollectionType()) {
+            CollectionType collectionType;
```

Comment:
How does this change the behavior? The only difference I see is that the `pa.getCollectionType()` is assigned to a variable after `fieldSpec` being cast to a more specific type. This is IMHO functionally equivalent to the current code.

Reply:
Before, `fieldSpec` was always cast to the `PluralAttribute` type. However, it was also possible that `fieldSpec` was of type `PluralQueryAttribute` which is - despite the similiar name - not a subclass of PluralAttribute which would result in a `ClassCastException` being thrown as mentioned in the original issue

Reply:
Yes, you are correct. I misunderstood the issue.